client/main.go

```diff
@@ -90,8 +71,23 @@ func NewClientConn(opt *ConnOptions) (*grpc.ClientConn, error) {
 		return nil, errors.Wrap(err, "parse cert")
 	}
 
-	creds := credentials.NewClientTLSFromCert(cp, "")
-	conn, err := grpc.Dial(opt.GRPCAuthority, grpc.WithTransportCredentials(creds))
+	return credentials.NewClientTLSFromCert(cp, ""), nil
+}
+
+func NewClientConn(opt *ConnOptions) (*grpc.ClientConn, error) {
+	method := grpc.WithInsecure()
+	if !opt.Insecure {
+		creds, err := fetchCert(opt.CertURL)
+		if err != nil {
+			return nil, err
+		}
+		method = grpc.WithTransportCredentials(creds)
+	}
+	conn, err := grpc.Dial(opt.GRPCAuthority,
+		method,
+		grpc.WithUnaryInterceptor(otelgrpc.UnaryClientInterceptor()),
+		grpc.WithStreamInterceptor(otelgrpc.StreamClientInterceptor()),
```

Comment:
excellent :)

Might want to add a client option to print the trace id in this case e.g. in the logs, so it's easy to take to a search box.

Reply:
Do you have an example elsewhere of how to do that I can crib from?